Source/WebCore/rendering/RenderElement.cpp

```diff
@@ -2118,7 +2118,8 @@ bool RenderElement::createsNewFormattingContext() const
         return true;
     return isInlineBlockOrInlineTable() || isFlexItemIncludingDeprecated()
         || isTableCell() || isTableCaption() || isFieldset() || isDocumentElementRenderer() || isRenderFragmentedFlow() || isSVGForeignObject()
-        || style().specifiesColumns() || style().columnSpan() == ColumnSpan::All || style().display() == DisplayType::FlowRoot || establishesIndependentFormattingContext();
+        || style().specifiesColumns() || style().columnSpan() == ColumnSpan::All || style().display() == DisplayType::FlowRoot || establishesIndependentFormattingContext()
+        || isSVGForeignObjectOrLegacySVGForeignObject();
```

Comment:
This is slightly incorrect as isSVGForeignObjectOrLegacySVGForeignObject() is supposed to create a new _block_ formatting context and this function is about _any_ formatting context root. I however think at this point we use it only in the context of BFC.

Reply:
I think short term, it is better to fix and wait till LBSE takeover because it will fix a lot more than just margin handling but for my learning, I am happy to learn more around it and would love any guidance and material to read on it.